### PR TITLE
Small navbar fixes

### PR DESCRIPTION
**Pull request by Richie78321**
* Move the `AuthButton` into the collapsible div so it scales properly on mobile.
* Make the Navbar sticky.
* Removed the tests for the `NavbarLink` because they are redundant with the tests for the `Navbar`.
* Updated the urls in the `Navbar` tests to reflect that actual urls.

### CODE REVIEW

File: src/main/webapp/js/common/react/tests/Navbar.test.js [hunk 1]
```diff
@@ -14,14 +14,19 @@ const navbarUrls = [
     name: 'Projects',
     requiresAuth: false,
   },
+  {
```
Comment (Sami-2000) [inline src/main/webapp/js/common/react/tests/Navbar.test.js:17]: Thanks for adding this!

Comment (Richie78321) [reply src/main/webapp/js/common/react/tests/Navbar.test.js:17]: Np! It's definitely better to have the test URLs match up with the production URLs anyways.

File: src/main/webapp/js/common/react/display_navbar.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ const urls = [
  */
 function initNavbar() {
   const navbarContainer = document.getElementById('navbar-container');
+  navbarContainer.classList.add('sticky-top');
```
Comment (billyeh) [inline src/main/webapp/js/common/react/display_navbar.js:40]: nice to learn about this class

Comment (Richie78321) [reply src/main/webapp/js/common/react/display_navbar.js:40]: It's super convenient, although it isn't yet [supported in all browsers](https://caniuse.com/#feat=css-sticky). The good news is that there's no real downside if the browser doesn't support it beyond the navbar not sticking to the top of the page.

**Pull request merged**